src/main/java/com/synopsys/integration/alert/web/config/FieldModelProcessor.java

```diff
@@ -0,0 +1,206 @@
+        this.configurationAccessor = configurationAccessor;
+        this.descriptorMap = descriptorMap;
+        this.fieldModelConverter = fieldModelConverter;
+    }
+
+    public FieldModel readFieldModel(ConfigurationModel configurationModel) throws AlertDatabaseConstraintException {
+        FieldModel fieldModel = convertToFieldModel(configurationModel);
+        final Optional<DescriptorActionApi> descriptorActionApi = retrieveDescriptorActionApi(fieldModel);
+        return descriptorActionApi.map(actionApi -> actionApi.readConfig(fieldModel)).orElse(fieldModel);
+    }
+
+    public FieldModel deleteFieldModel(ConfigurationModel configurationModel) throws AlertDatabaseConstraintException {
+        final FieldModel fieldModel = convertToFieldModel(configurationModel);
+        final Optional<DescriptorActionApi> descriptorActionApi = retrieveDescriptorActionApi(fieldModel);
+        return descriptorActionApi.map(actionApi -> actionApi.deleteConfig(fieldModel)).orElse(fieldModel);
+    }
+
+    public Map<String, ConfigurationFieldModel> saveFieldModel(FieldModel fieldModel) throws AlertDatabaseConstraintException {
+        final Optional<DescriptorActionApi> descriptorActionApi = retrieveDescriptorActionApi(fieldModel);
+        FieldModel modelToSave = descriptorActionApi.map(actionApi -> actionApi.saveConfig(fieldModel)).orElse(fieldModel);
+        return fieldModelConverter.convertFromFieldModel(modelToSave);
+    }
+
+    public void validateFieldModel(FieldModel fieldModel, Map<String, String> fieldErrors) {
+        final Optional<DescriptorActionApi> descriptorActionApi = retrieveDescriptorActionApi(fieldModel);
+        if (descriptorActionApi.isPresent()) {
+            final Map<String, ConfigField> configFields = retrieveUIConfigFields(fieldModel.getContext(), fieldModel.getDescriptorName())
+                                                              .stream()
+                                                              .collect(Collectors.toMap(ConfigField::getKey, Function.identity()));
+            descriptorActionApi.get().validateConfig(configFields, fieldModel, fieldErrors);
+        }
+    }
+
+    public String testFieldModel(FieldModel fieldModel, String destination) throws IntegrationException {
+        final Optional<DescriptorActionApi> descriptorActionApi = retrieveDescriptorActionApi(fieldModel);
+        if (descriptorActionApi.isPresent()) {
+            final DescriptorActionApi descriptorApi = descriptorActionApi.get();
+            final TestConfigModel testConfig = descriptorApi.createTestConfigModel(fieldModel, destination);
+            // TODO look to see if FieldAccessor can be modified to allow mutable maps. FieldAccessor and destination can then be passed here
+            descriptorApi.testConfig(testConfig);
+            return "Successfully sent test message.";
+        } else {
+            return "Descriptor action api did not exist: " + fieldModel.getDescriptorName();
```

Comment:
There is similar code elsewhere to do this. I think it makes sense to have this. It may not make sense to a User but if our testing is good they should never see this and it make development a lot easier

Reply:
I would maybe change the message to "Failed to send test message:  Failed to find a component to process the request."
maybe change component to descriptor in the message. 
In general I don't like the idea of showing text in the UI that references descriptor or action api since those are server concepts in the product.  Might be leaking information about the internals.  A log message would be ok,  but not a message that shows up in the UI.  If it is implemented correctly it is a message they will never see.